src/features/canvas/CanvasIntegration.ts

```diff
@@ -2,10 +2,9 @@
 // Initialize Canvas polling
 const CANVAS_API = 'https://sit.instructure.com/api/v1/';
 function getUpcomingAssignments() {
-    chrome.storage.local.get('tokens', tokens => {
-        // If you want to test this, uncomment the code below with your token
-        // tokens.canvas = '<YOUR TOKEN>';
-        if (!tokens.canvas) {
+    chrome.storage.local.get('tokens', object => {
+        let {tokens} = object;
+        if (tokens.canvas && tokens.canvas !== '') {
```

Comment:
Empty string is falsy in js
```suggestion
        if (tokens.canvas) {
```